tests/common.go

```diff
@@ -4595,6 +4606,70 @@ func CreateBucket(provider string, bucketName string) {
 	})
 }
 
+// IsBucketEmpty returns true if the bucket for a provider is empty
+func IsBucketEmpty(provider, bucketName string) (bool, error) {
+	switch provider {
+	case drivers.ProviderAws:
+		result, err := IsS3BucketEmpty(bucketName)
+		if err != nil {
+			return result, err
+		}
+		return result, err
+	case drivers.ProviderNfs:
+		result, err := IsNFSSubPathEmpty(bucketName)
+		if err != nil {
+			return result, err
+		}
+		return result, err
+	default:
+		return false, fmt.Errorf("function does not support %s provider", provider)
+	}
+}
+
+func IsNFSSubPathEmpty(subPath string) (bool, error) {
+	// Get NFS share details from ENV variables.
+	creds := GetNfsInfoFromEnv()
+	mountDir := fmt.Sprintf("/tmp/nfsMount" + RandomString(4))
+
+	// Mount the NFS share to the master node.
+	masterNode := node.GetMasterNodes()[0]
+	mountCmds := []string{
+		fmt.Sprintf("mkdir -p %s", mountDir),
+		fmt.Sprintf("mount -t nfs %s:%s %s", creds.NfsServerAddress, creds.NfsPath, mountDir),
+		fmt.Sprintf("find %s/%s -type f", mountDir, subPath),
+	}
+	for _, cmd := range mountCmds {
+		output, err := runCmdGetOutput(cmd, masterNode)
+		log.FailOnError(err, fmt.Sprintf("Failed to run [%s] command on node [%s], error : [%s]", cmd, masterNode, err))
+		log.Infof("Output from command [%s] -\n%s", cmd, output)
+	}
+
+	defer func() {
+		// Unmount the NFS share from the master node.
+		umountCmds := []string{
+			fmt.Sprintf("umount %s", mountDir),
+			fmt.Sprintf("rm -rf %s", mountDir),
+		}
+		for _, cmd := range umountCmds {
+			err := runCmd(cmd, masterNode)
+			log.FailOnError(err, fmt.Sprintf("Failed to run [%s] command on node [%s], error : [%s]", cmd, masterNode, err))
+		}
+	}()
+
+	// List the files in subpath from NFS share path.
+	log.Infof("Checking the contents in NFS share subpath: [%s] from path: [%s] on server: [%s]", subPath, creds.NfsPath, creds.NfsServerAddress)
+	fileCountCmd := fmt.Sprintf("find %s/%s -type f | wc -l", mountDir, subPath)
```

Comment:
Are we checking file count here or just directory exists not ? Otherwise, find command succeeds if directory exists and no files exists in that directory.

Reply:
We are just checking the file count which is why I have added `-type f`. Directories can exist.

Reply:
@mkoppal-px Please see this:

trenukarya--Mac14:repos trenukarya$ find ~/work/repos/torpedo/ -type f | wc -l
   11198
trenukarya--Mac14:repos trenukarya$ find ~/work/temp/ -type f | wc -l
       0
trenukarya--Mac14:repos trenukarya$ echo $?
0

Reply:
Did not get you here? We are checking for the total number of files present in the subpath

Reply:
Clarified this issue in-person